gensim/models/doc2vec.py

```diff
@@ -635,8 +635,13 @@ def scan_vocab(self, documents, progress_per=10000, trim_rule=None):
         min_reduce = 1
         interval_start = default_timer() - 0.00001  # guard against next sample being identical
         interval_count = 0
+        checked_string_types = 0
         vocab = defaultdict(int)
         for document_no, document in enumerate(documents):
+            if not checked_string_types:
```

Comment:
How fast / slow is `warnings.warn`?

If the check is fast (before argument formatting), it may be easier to just issue `warnings.warn` and let the warnings module handle print-never/print-once/print-every-time internally.